de-lib/src/main/java/org/iplantc/de/theme/base/client/diskResource/grid/cells/DiskResourceDotMenuDefaultAppearance.java

```diff
@@ -0,0 +1,112 @@
+
+    private IplantResources iplantResources;
+    private IplantDisplayStrings iplantDisplayStrings;
+    private DiskResourceMessages diskResourceMessages;
+    private AppDotMenuDefaultAppearance appDotMenuDefaultAppearance;
+    private DiskResourceDotMenuDisplayStrings displayStrings;
+
+    public DiskResourceDotMenuDefaultAppearance() {
+        this(GWT.<IplantResources>create(IplantResources.class),
+             GWT.<IplantDisplayStrings>create(IplantDisplayStrings.class),
+             GWT.<DiskResourceMessages>create(DiskResourceMessages.class),
+             GWT.<AppDotMenuDefaultAppearance>create(AppDotMenuDefaultAppearance.class),
+             GWT.<DiskResourceDotMenuDisplayStrings>create(DiskResourceDotMenuDisplayStrings.class));
+    }
+
+    public DiskResourceDotMenuDefaultAppearance(IplantResources iplantResources,
+                                                IplantDisplayStrings iplantDisplayStrings,
+                                                DiskResourceMessages diskResourceMessages,
+                                                AppDotMenuDefaultAppearance appDotMenuDefaultAppearance,
+                                                DiskResourceDotMenuDisplayStrings displayStrings) {
+
+        this.iplantResources = iplantResources;
+        this.iplantDisplayStrings = iplantDisplayStrings;
+        this.diskResourceMessages = diskResourceMessages;
+        this.appDotMenuDefaultAppearance = appDotMenuDefaultAppearance;
+        this.displayStrings = displayStrings;
+    }
+
+    @Override
+
+    public void render(Cell.Context context, DiskResource value, SafeHtmlBuilder sb, String debugId) {
+        appDotMenuDefaultAppearance.render(context, sb, debugId);
+    }
+
+    @Override
+    public ImageResource dotMenuIcon() {
+        return appDotMenuDefaultAppearance.dotMenuIcon();
+    }
+
+    @Override
+    public String favoriteText(DiskResource diskResource) {
+        if (diskResource.isFavorite()) {
+            return iplantDisplayStrings.remAppFromFav();
+        } else {
+            return iplantDisplayStrings.addAppToFav();
+        }
+    }
+
+    @Override
+    public ImageResource favoriteIcon(DiskResource diskResource) {
+        if (diskResource.isFavorite()) {
+            return iplantResources.favIconDelete();
+        } else {
+            return iplantResources.favIconAdd();
+        }
+    }
+
+    @Override
+    public String commentText() {
+        return iplantDisplayStrings.comments();
+    }
+
+    @Override
+    public ImageResource commentIcon() {
+        return iplantResources.userComment();
+    }
+
+    @Override
+    public String dataLinkText() {
+        return displayStrings.dataLinkText();
+    }
+
+    @Override
+    public ImageResource dataLinkIcon() {
+        return iplantResources.linkIcon();
+    }
+
+    @Override
+    public String shareText() {
+        return diskResourceMessages.share();
```

Comment:
I think this string is a change, and this previously said "Share with Collaborators..." (as opposed to "share via public link" or similar). I don't know that I see this as a problematic change but I didn't see it mentioned so I wanted to ask if it was intentional.

(I also notice the public link one changed, but it seems the previous text was "Share Path", which is less than useful, so...)

Reply:
It should probably be "Share with Collaborators..." to be consistent with the menu, so I'll change that.  The public link one is slightly different than the menu item since the menu item brings up the full editing links view whereas this either creates a public link or fetches a public link and shows it so... I'm not sure.  Maybe it'd be better to have the text match and update this so they show the same view?

Reply:
I think it makes sense to keep the public links as two different things (as I mentioned I think I forgot the more shortcut-style simple dialog existed, heh). I guess for this exact line it makes sense to have the menu and this match, if they bring up the same dialog, though?

Reply:
Yeah, if they bring up the same thing, then the text should match, I agree.  OK, I'll keep the shortcut link dialog and keep the text slightly different then for this case.